src/main/java/org/folio/auth/authtokenmodule/tokens/Token.java

```diff
@@ -206,46 +256,9 @@ protected void validateCommon(HttpServerRequest request) throws TokenValidationE
     }
   }
 
-  public static Token parse(String sourceToken, TokenCreator tokenCreator) throws TokenValidationException {
-    JsonObject claims;
-    final String invalidTokenMsg = "Invalid token";
-
-    try {
-      if (isEncrypted(sourceToken)) {
-        String tokenContent = tokenCreator.decodeJWEToken(sourceToken);
-        claims = new JsonObject(tokenContent);
-      } else {
-        tokenCreator.checkJWTToken(sourceToken);
-        claims = getClaims(sourceToken);
-      }
-    } catch (ParseException p) {
-      throw new TokenValidationException(invalidTokenMsg, p, 401);
-    } catch (JOSEException j) {
-      throw new TokenValidationException(invalidTokenMsg, j, 401);
-    } catch (BadSignatureException b) {
-      final String msg = "Invalid token signature. "
-          + "This might have been caused by a mod-authtoken restart if jwt.signing.key is not set, "
-          + "or by running multiple mod-authtoken instances without setting the same jwt.signing.key.";
-      throw new TokenValidationException(msg, b, 401);
-    }
-
-    String tokenType = claims.getString("type");
-    if (tokenType == null)
-      throw new TokenValidationException("Token has no type", 400);
-
-    switch (tokenType) {
-      case AccessToken.type:
-        return new AccessToken(sourceToken, claims);
-      case RefreshToken.type:
-        return new RefreshToken(sourceToken, claims);
-      case ApiToken.type:
-        return new ApiToken(sourceToken, claims);
-      case DummyToken.type:
-        return new DummyToken(sourceToken, claims);
-      case ModuleToken.type:
-        return new ModuleToken(sourceToken, claims);
-      default:
-        throw new TokenValidationException("Unable to parse token", 400);
-    }
+  protected boolean tokenHasExpired(JsonObject tokenClaims) {
```

Comment:
If it is not static it should check the `protected JsonObject claims` of this Token.
If it takes a toeknClaims parameter it should be static.

Reply:
Agreed. It's an instance method and has access to the claims. Removing param.